client/web/src/enterprise/batches/DropdownButton.tsx

```diff
@@ -0,0 +1,196 @@
+import classNames from 'classnames'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+
+import styles from './DropdownButton.module.scss'
+
+export interface Action {
+    /* The type of action. Used internally. */
+    type: string
+    /* The button label for the action. */
+    buttonLabel: string
+    /* The title in the dropdown menu item. */
+    dropdownTitle: string
+    /* The description in the dropdown menu item. */
+    dropdownDescription: string
+    /**
+     * Invoked when the action is triggered. Either onDone or onCancel need to
+     * be called eventually. Can return a JSX.Element to be rendered adjacent to
+     * the button (i.e. a modal).
+     */
+    onTrigger: (onDone: () => void, onCancel: () => void) => Promise<void | JSX.Element> | void | JSX.Element
+    /** If set, displays an experimental badge next to the dropdown title. */
+    experimental?: boolean
+}
+
+export interface Props {
+    actions: Action[]
+    defaultAction?: number
+    disabled?: boolean
+    dropdownMenuPosition?: 'left' | 'right'
+    initiallyOpen?: boolean
+    onLabel?: (label: string | undefined) => void
+    placeholder?: string
+    tooltip?: string
+}
+
+export const DropdownButton: React.FunctionComponent<Props> = ({
+    actions,
+    defaultAction,
+    disabled,
+    dropdownMenuPosition,
+    initiallyOpen,
+    onLabel,
+    placeholder,
+    tooltip,
+}) => {
+    placeholder ??= 'Select action'
```

Comment:
This seems a bit simpler

```suggestion
    placeholder = 'Select action',
    tooltip,
}) => {
```